Disable adjustment and hyphenation (local change).
Set .Dt and .Xr in roman, not italic font (local fix for 1.23 regressions).
Set literal text in CR, not bold font (local fix for 1.23 regression).
Map hyphen-minus and accents to ASCII in the traditional way (local change).
Index: tmac/mdoc.local
--- tmac/mdoc.local.orig
+++ tmac/mdoc.local
@@ -12,6 +12,34 @@
 .\" You might prefer to remap it to bold ("B") instead.
 .\" .if n .ftr CW R
 .\"
+.\" ----------------------------------------------------------------------
+.\" OpenBSD-specific changes
+.\" ----------------------------------------------------------------------
+.\" Disable adjustment and hyphenation by default,
+.\" such that manuals look similar with groff and mandoc(1).
+.ad l
+.de ad
+..
+.nh
+.de hy
+.nh
+..
+.\" Fix regressions in .Dt and .Xr formatting
+.ds doc-page-topic-font \fR
+.ds doc-Xr-font \fR
+.\" Fix regressions in .Bd, .Bf, .Dl, and .Li formatting
+.ds doc-Li-font \f(CR
+.\"
+.\" Map ASCII input single quotes and hyphen-minus to ASCII output
+.\" until we properly escape them in OpenBSD manual pages.
+.if '\*[.T]'utf8' \{\
+.  char ' \N'39'
+.  char - \N'45'
+.  char ^ \N'94'
+.  char ` \N'96'
+.  char ~ \N'126'
+.\}
+.\"
 .\" Local Variables:
 .\" mode: nroff
 .\" fill-column: 72
